pkg/platform/controller/cluster/deletion/cluster_deleter.go

```diff
@@ -153,6 +153,17 @@ func (d *clusterDeleter) deleteCluster(ctx context.Context, cluster *platformv1.
 	if err != nil && !errors.IsNotFound(err) {
 		return err
 	}
+
+	if cluster.Spec.ClusterCredentialRef == nil {
+		log.FromContext(ctx).Errorf("invalid ClusterCredentialRef, cls: %s", cluster.Name)
+		return nil
+	}
+	err = d.platformClient.ClusterCredentials().Delete(ctx, cluster.Spec.ClusterCredentialRef.Name, metav1.DeleteOptions{})
+	if err != nil && !errors.IsNotFound(err) {
```

Comment:
If we return err, it will retry full process of cluster deletion, but cluster is already deleted, so this retry always failes.